site/content/extend/plugins/developer-workflow.md

```diff
@@ -0,0 +1,66 @@
+---
+title: "Developer Workflow"
+date: 2020-07-11T23:00:00-04:00
+weight: 3
+subsection: Plugins
+---
+
+### Common `make` commands
+
+- `make dist` - Compile the plugin into a g-zipped file, ready to upload to a Mattermost server. The file is saved in the plugin repo's `dist` folder.
+- `make deploy` - Compiles the plugin using the `make dist` command, then automatically deploys the plugin to the Mattermost server
+- `make watch` - Uses webpack's watch feature to re-compile and deploy the webapp portion of your plugin on any change to the `webapp/src` folder
+- `make test` - Runs the plugin's server tests and webapp tests
+- `make check-style` - Runs linting checks on the plugin's server and webapp folders
+- `make enable` - Enables the plugin on the Mattermost server
+- `make disable` - Disables the plugin on the Mattermost server
+- `make reset` - Disables and re-enables the plugin on the Mattermost server
+- `make debug-plugin` - Starts a `delve` process and attaches it to your running plugin
+
+You can run the development build of the plugin by setting the environment variable `MM_DEBUG=1`, or prefixing the variable at the beginning of the `make` command. For example, `MM_DEBUG=1 make deploy` will deploy the development build of the plugin to your server, allowing you to have a more fluid debugging experience. To instead use the production build of the plugin, unset the `MM_DEBUG` environment variable before running the `make` commands.
+
+### Exposing the Mattermost server using `ngrok`
+
+When a public integrates with an external service, webhooks and/or authentication redirects are necessary, which requires your local server to be available on the web. In order for your Mattermost server to be available to process webhook requests, it needs to expose its port to an external address. A common way to do this is to use the command line tool [ngrok](https://ngrok.com). Follow these steps to set up `ngrok` with your server:
+
+- Download the `ngrok` tool from [here](https://ngrok.com/download)
```

Comment:
```suggestion
- Download the ngrok tool from [here](https://ngrok.com/download).
```

Reply:
I feel like it makes sense to have "ngrok" in the code quotes. It's a technical product that stands out when formatted using the code format. I'll change if you still think it should be changed :+1:

Reply:
I agree that having it stand out is useful especially as it's a lowercase product name. That said, the code format is used for commands/fields/values and so on and having ngrok in code format implies it's one of those ([as per the style guide](https://handbook.mattermost.com/operations/operations/company-processes/publishing/publishing-guidelines/voice-tone-and-writing-style-guidelines/documentation-style-guide#text-formatting)). I'm 5/5 on this but open to suggestions around making it stand out more (bearing in mind that any formatting applied to this product name would need to be applied to all other products/tools we mention in our docs).

Reply:
`ngonk` is both a product name and a command. Given that your audience here are developers who use `ngonk` as a command via a shell, it makes sense to mark is as such.

Reply:
@justinegeffen What's your opinion on this given Ben's input above?

Reply:
That makes sense, I'm happy to go with that.